app/controllers/RestaurantController.scala

```
@@ -0,0 +1,25 @@
+package controllers
+
+import javax.inject.Inject
+import persistence.models.Formats._
+import play.api.libs.json._
+import play.api.mvc.{Action, AnyContent, Controller}
+import services.RestaurantService
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class RestaurantController @Inject()(restaurantsService: RestaurantService) extends Controller {
+
+  def listRestaurants(): Action[AnyContent] = Action.async { request =>
+    restaurantsService.getAllRestaurants.map(restaurants =>
+      Ok(Json.toJson(restaurants))
+    )
+  }
+
+  def createRestaurant() = Action(parse.json) { requset =>
+    val json = Json.fromJson(requset.body)
+    try {
```

you want to return an error back to the end-user, not only log it?